### PR TITLE
Feature/task 3582 increase risk calculation cycle

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
Implement 4 hour cycle depending on app config.

## Link to Jira
[3582](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-3582)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/Home/HomeInteractor.swift [hunk 1]
```diff
@@ -229,7 +229,7 @@ extension HomeInteractor {
 		riskLevelConfigurator = nil
 		inactiveConfigurator = nil
 
-		let detectionInterval = (riskProvider.configuration.exposureDetectionInterval.day ?? 1) * 24
+        let detectionInterval = riskProvider.riskProvidingConfiguration.exposureDetectionInterval.hour ?? RiskProvidingConfiguration.defaultExposureDetectionsInterval
```
Comment (SwissMountainFrog) [inline src/xcode/ENA/ENA/Source/Scenes/Home/HomeInteractor.swift:232]: Maybe too niggling 😸  Just a missing inline tab (code styling)

**Review by SwissMountainFrog — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/RiskProvidingConfiguration.swift [hunk 2]
```diff
@@ -21,6 +21,18 @@ import Foundation
 
 /// Used to configure a `RiskLevelProvider`.
 struct RiskProvidingConfiguration {
+
+    static let defaultExposureDetectionsInterval = 24 / defaultMaxExposureDetectionsPerInterval
+    private static let defaultMaxExposureDetectionsPerInterval = 1
+
+	static var defaultConfiguration: RiskProvidingConfiguration {
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/RiskProvidingConfiguration.swift:28]: _Swiftier_ name:
```suggestion
	static var default: RiskProvidingConfiguration {
```

If accepted: needs also adjustments to code above!

File: src/xcode/ENA/ENA/Source/AppDelegate.swift [hunk 3]
```diff
@@ -141,7 +133,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 		)
 
 		return RiskProvider(
-			configuration: config,
+			configuration: RiskProvidingConfiguration.defaultConfiguration,
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/AppDelegate.swift:136]: _Swiftier_ naming:
```suggestion
			configuration: .default,
```

If accepted: needs also adjustments to code below!

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift [hunk 4]
```diff
@@ -433,6 +437,26 @@ extension RiskProvider: RiskProviding {
...
+
+    private func updateRiskProvidingConfiguration(with appConfig: SAP_Internal_ApplicationConfiguration) {
+        let maxExposureDetectionsPerInterval = Int(appConfig.iosExposureDetectionParameters.maxExposureDetectionsPerInterval)
+
+        var exposureDetectionInterval: DateComponents
+        if maxExposureDetectionsPerInterval == 0 {
+            // Deactivate exposure detection by setting a high, not reachable value.
+            // Unfortunately Int.max does not work. It leads to DateComponents.hour == nil.
+            let fiftyYears = 24 * 365 * 50
+            exposureDetectionInterval = DateComponents(hour: fiftyYears)
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift:449]:
```suggestion
            exposureDetectionInterval = DateComponents(hour: Int.max.advanced(by: -1)) // a.k.a. 1 BER build
```

**Review by whiskey — COMMENTED** [review]
Just 2-3 comments. Looks generally ok 👍

File: src/xcode/ENA/ENA/Source/Scenes/Home/HomeInteractor.swift (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/Home/HomeInteractor.swift:232]: No problem. But it looks ok ob my side. No missing tab 😇
I tried to fix it by removing and adding the tab again 😅

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/RiskProvidingConfiguration.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/RiskProvidingConfiguration.swift:28]: I would also prefer "default", but its a keyword.
This works "`default`" ☺️

File: src/xcode/ENA/ENA/Source/AppDelegate.swift (hunk 3 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/AppDelegate.swift:136]: done

File: src/xcode/ENA/ENA/Source/Scenes/Home/HomeInteractor.swift (hunk 1 above)
Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Scenes/Home/HomeInteractor.swift]: Nobody likes tabs. We should use white spaces 🙃
Github has issues with displaying this. We can mostly ignore it. I would bet `swiftlint` will find any misaligned code.

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/RiskProvidingConfiguration.swift (hunk 2 above)
Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/RiskProvidingConfiguration.swift]: 🤦‍♂️ true…

**Review by whiskey — APPROVED** [review]

**Pull request merged**